func.go

```diff
@@ -163,6 +164,16 @@ func RegisterFunc(fptr interface{}, cfn uintptr) {
 				panic("purego: unsupported kind " + arg.Kind().String())
 			}
 		}
+		if ty.NumOut() > 0 && ty.Out(0).Kind() == reflect.Struct {
+			if runtime.GOOS != "darwin" {
+				panic("purego: struct return values only supported on darwin arm64 & amd64")
+			}
```

Comment:
Why do you check for function having more than 0 return parameters, while at line 244 for precise equivalence to one?

Imo, since right now (and likely forever) we don't support more than 1 return parameter, it'd be better to be stricter about it and panic with a corresponding message, like `purego:  functions may return only one value`

Reply:
Checking for 1 is probably more clear. No need to panic here since it's already done at L115